src/views/StoryViewer.vue

```diff
@@ -19,31 +19,76 @@ export default {
   name: 'StoryViewer',
   data: () => {
     return {
-      frameDuration: 2000,
-      storyEnd: false
+      frameDuration: 6000,
+      frameStarting: null,
+      frameRemaining: null,
+      storyEnd: false,
+      currentTimeout: null,
+      isPaused: false
     }
   },
   components: {
     ImageAttribution
   },
-  computed: mapGetters(['currentFrame', 'storyLength']),
+  computed: {
+    ...mapGetters(['currentFrame', 'storyLength']),
+    animationDuration: function () {
+      return this.frameDuration / 1000 + 's'
+    }
+  },
   methods: {
     ...mapActions(['selectFrame']),
+    setFrameTimeout: function(f) {
+      const duration = this.frameRemaining ? this.frameRemaining : this.frameDuration
+      this.frameStarting = Date.now()
+      this.currentTimeout = setTimeout( () => {
+        f()
+        clearTimeout(this.currentTimeout)
```

Comment:
Note that you don't have to clear the timeout inside of the handler. `clearTimeout` is to cancel the handler, now it's too late to cancel it since it's already running.